rotkehlchen/chain/evm/decoding/kyber/decoder.py

```diff
@@ -0,0 +1,183 @@
+
+KYBER_TRADE_LEGACY = b'\xf7$\xb4\xdff\x17G6\x12\xb5=\x7f\x88\xec\xc6\xea\x980t\xb3\t`\xa0I\xfc\xd0e\x7f\xfe\x80\x80\x83'  # noqa: E501
+KYBER_TRADE = b"\xd3\x0c\xa3\x99\xcbCP~\xce\xc6\xa6)\xa3\\\xf4^\xb9\x8c\xdaU\x0c'im\xcb\r\x8cJ8s\xcel"  # noqa: E501
+KYBER_LEGACY_CONTRACT = string_to_evm_address('0x9ae49C0d7F8F9EF4B864e004FE86Ac8294E20950')
+KYBER_LEGACY_CONTRACT_MIGRATED = string_to_evm_address('0x65bF64Ff5f51272f729BDcD7AcFB00677ced86Cd')  # noqa: E501
+KYBER_LEGACY_CONTRACT_UPGRADED = string_to_evm_address('0x9AAb3f75489902f3a48495025729a0AF77d4b11e')  # noqa: E501
+KYBER_AGGREGATOR_CONTRACT: Final = string_to_evm_address('0x6131B5fae19EA4f9D964eAc0408E4408b66337b5')  # noqa: E501
+KYBER_AGGREGATOR_TRADE: Final = b'\xd6\xd4\xf5h\x1c$l\x9fB\xc2\x03\xe2\x87\x97Z\xf1`\x1f\x8d\xf8\x03Z\x92Q\xf7\x9a\xab\\\x8f\t\xe2\xf8'  # noqa: E501
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _maybe_update_events_legacy_contrats(
+        decoded_events: list['EvmEvent'],
+        sender: ChecksumEvmAddress,
+        source_asset: CryptoAsset,
+        destination_asset: CryptoAsset,
+        spent_amount: FVal,
+        return_amount: FVal,
+        notify_user: Callable[['EvmEvent', str], None],
+) -> None:
+    """
+    Use the information from a trade transaction to modify the HistoryEvents from receive/send to
+    trade if the conditions are correct.
+    """
+    in_event = out_event = None
+    for event in decoded_events:
+        try:
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+        except (UnknownAsset, WrongAssetType):
+            notify_user(event, CPT_KYBER)
+            continue
+        # it can happen that a spend event get decoded first by an amm decoder. To make sure that
+        # the event matches we check both event type and subtype
+        if (event.event_type == HistoryEventType.SPEND or event.event_subtype == HistoryEventSubType.SPEND) and event.location_label == sender and event.asset == source_asset and event.balance.amount == spent_amount:  # noqa: E501
+            event.event_type = HistoryEventType.TRADE
+            event.event_subtype = HistoryEventSubType.SPEND
+            event.counterparty = CPT_KYBER
+            event.notes = f'Swap {event.balance.amount} {crypto_asset.symbol} in kyber'
+            out_event = event
+        elif (event.event_type == HistoryEventType.RECEIVE or event.event_subtype == HistoryEventSubType.RECEIVE) and event.location_label == sender and event.balance.amount == return_amount and destination_asset == event.asset:  # noqa: E501
+            event.event_type = HistoryEventType.TRADE
+            event.event_subtype = HistoryEventSubType.RECEIVE
+            event.counterparty = CPT_KYBER
+            event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} from kyber swap'
+            in_event = event
+
+        maybe_reshuffle_events(ordered_events=[out_event, in_event], events_list=decoded_events)
```

Comment:
before doing the reorg make sure that in_event and out_event aren't none. It could be that for some reason one of the events didn't get properly decoded